web/src/echo.ts

```diff
@@ -61,28 +131,43 @@ function show_message_failed(message_id, failed_msg) {
     });
 }
 
-function show_failed_message_success(message_id) {
+function show_failed_message_success(message_id: number): void {
     // Previously failed message succeeded
     message_live_update.update_message_in_all_views(message_id, ($row) => {
         $row.find(".message_failed").toggleClass("hide", true);
     });
 }
 
-function failed_message_success(message_id) {
-    message_store.get(message_id).failed_request = false;
+function failed_message_success(message_id: number): void {
+    message_store.get(message_id)!.failed_request = false;
     show_failed_message_success(message_id);
 }
 
-function resend_message(message, $row, {on_send_message_success, send_message}) {
-    message.content = message.raw_content;
+function resend_message(
+    message: Message,
+    $row: JQuery,
+    {
+        on_send_message_success,
+        send_message,
+    }: {
+        on_send_message_success: (request: Message, data: Data) => void;
+        send_message: (
+            request: Message,
+            on_success: (data: unknown) => void,
+            error: (response: string, _server_error_code: string) => void,
+        ) => void;
+    },
+): void {
+    message.content = message.raw_content!;
```

Comment:
This is pretty weird code; I wonder if there's a nicer way to do the types for it.